src/containerbuddy/discovery.go

```diff
@@ -4,4 +4,5 @@ type DiscoveryService interface {
 	SendHeartbeat(*ServiceConfig)
 	CheckForUpstreamChanges(*BackendConfig) bool
 	MarkForMaintenance(*ServiceConfig)
+	Deregister(*ServiceConfig)
```

Comment:
I like that you've extended the `DiscoveryService` such that if `MarkForMaintenance` and `Deregister` need to be separate implementations for a non-Consul discovery backend we can do that easily.